### ISSUE TITLE
Notifcations are only “You have a new message”

### ISSUE THREAD

**Issue by Kamik423** [TelegramMessenger/Telegram-iOS#833]
### Checklist
- [x] I am reporting an issue in existing functionality that does not work as intended
- [x] I've searched for existing [GitHub issues](https://github.com/telegrammessenger/Telegram-iOS/issues)

### Description
Telegram censors notification text even though this feature is disabled. This started occurring a few days ago on my iPad. I did not touch any settings but all messages get censored now. The same is not true for my iPhone signed into the same account with the same settings.

### Expected Behavior
The message content should be in the notification.

### Actual Behavior
It just says “You have a new message”

### Steps to Reproduce
This is not with due to the iOS notification preview censoring since that changes the message to “Message” instead of “You have a new message”. So the issue is due to Telegram internal settings. In Telegram itself the “Message Preview” switch is also enabled (I tried toggling it off and on) for all chats, groups, and channels.

This issue might be related to [this bug report](https://bugs.telegram.org/c/174) although for me this only started happening a few days ago — potentially with the release of the latest AppStore version of the app.

### Screenshots
![A12FD1A5-D29C-4F2F-9974-C8A0CD65995F](https://user-images.githubusercontent.com/16278233/188312740-00fe5b20-e30e-4833-8330-a83018b98d69.jpeg)
![25122307-4713-45B8-8D0F-6B4AAC5E4666](https://user-images.githubusercontent.com/16278233/188312743-c11d1765-d97b-43a7-94bb-f9fe9749b6a9.jpeg)
![7D079E07-DA73-451D-BF68-72296F61860F](https://user-images.githubusercontent.com/16278233/188312746-6ace5371-b617-4451-ba9f-ce3f5bc34d2d.jpeg)

### Environment
**Device:** `iPad 12.9 with FaceID first generation`

**iOS version**: `16.1 beta`

**App version:** `8.9.1`

**Comment by JhihPong** [comment]
I have the same question too. Do you have any ideas to solve this problem?

**Comment by Kamik423** [comment]
I am not an expert or a developer on this project, but to me this statement

https://github.com/TelegramMessenger/Telegram-iOS/blob/7d335e817a5a67214fa968075f108ba6aad03e37/Telegram/NotificationService/Sources/NotificationService.swift#L516-L520

looks the wrong way around. In my option the locked message should be set if the messages are empty – not un-empty.

**Comment by abumora** [comment]
do you have a solution?

**Comment by skiks** [comment]
Same problem! Already 2 months! Empty notifications. its drain my battery so much!
![IMG_7354](https://github.com/TelegramMessenger/Telegram-iOS/assets/26214997/4e51c7b3-b8f8-4f77-b6df-8af8105d8dca)

**Comment by Erik262** [comment]
Now, in 2024, this issue still persists.!!

**Comment by farzadso** [comment]
Yeah, still happening to me on all my devices.

**Comment by Erik262** [comment]
Yeah, I feel the same way. I can't believe that we're the only ones dealing with this problem.

**Comment by farzadso** [comment]
I just reset my notification settings (such a pain to mute all those chats one by one again) and it seems to be fixed now.

**Comment by Erik262** [comment]
> I just reset my notification settings (such a pain to mute all those chats one by one again) and it seems to be fixed now.

Did you do this inside the app or in your phones settings?

**Comment by farzadso** [comment]
> > I just reset my notification settings (such a pain to mute all those chats one by one again) and it seems to be fixed now.
> 
> Did you do this inside the app or in your phones settings?

In my phone settings.

**Comment by zanyar3** [comment]
same issue 2024 yet !

**Comment by shayangha** [comment]
I have the same problem too

**Comment by shayangha** [comment]
sometimes when I restart my phone fixed for me.

**Comment by Erik262** [comment]
Yeah, same here. Sometimes the text is displayed, and sometimes it is not... No matter what, Telegram is prioritizing selling emojis instead of addressing bugs, even years before introducing any subscription "feature."

**Comment by jjaychen1e** [comment]
Same issue here. And these notifications drained my battery.

**Comment by Kaiser-DMr** [comment]
到现在为止还有这个错误

**Comment by Yuisyuu** [comment]
#428

**Comment by nyg** [comment]
Same issue!

**Comment by Paragkk** [comment]
Is there any solution yet. it started after upgrading to ios 18.0.1 on my iphone 15 pro max

**Comment by DeadNumbers** [comment]
The same problem, 16Pro Max, IOS 18.2

**Comment by Erik262** [comment]
The problem still persists. I’ll see you in 2026!

**Comment by chechoora** [comment]
This issue occurs because the app lacks the com.apple.developer.usernotifications.filtering entitlement. The original Telegram iOS app uses this entitlement to filter out service-related APNS messages (like badge counter updates and old notification cleanup). Without this entitlement, you'll receive all these background notification updates. Check the [Apple documentation](https://developer.apple.com/documentation/bundleresources/entitlements/com.apple.developer.usernotifications.filtering) for more details on implementing this entitlement.

Right now I am trying to figure out workaround but no luck.

**Comment by nasaboy** [comment]
The same problem, iPhone15, IOS 18.2.1.App version  11.6.3

**Comment by ariaieboy** [comment]
same here in 2025

**Comment by nice01166** [comment]
This problem still exists and drains my phone's battery power

**Comment by kaizenGG** [comment]
same in 2025, fucking shame, phone hot and draining the fucking battery.

**Comment by RaveyFok** [comment]
You have a new message，现在2025 年了依然存在

**Comment by Y-Yy88** [comment]
我想问一下，为什么我这里没有任何通知呢？除非在我打开它的时候。

**Comment by Erik262** [comment]
> The problem still persists. I’ll see you in 2026!

hahahahahahahaha well see ya in 2027 again xD

**Comment by nasaboy** [comment]
> > The problem still persists. I’ll see you in 2026!
> 
> hahahahahahahaha well see ya in 2027 again xD

LOL

**Comment by Wisil** [comment]
2026年了，这个问题依然存在。。。

**Comment by j4hangir** [comment]
Since two days ago been the same, I suspect after using iPhone Mirroring.

Tried everything, updating, rebooting, resetting notification settings, all the same, the body is always "You have a new message".

**Comment by j4hangir** [comment]
Note, this issue differs from #2037 as here it's a generic message from Telegram, but there it's an actual notification from contacts but the body is unset.

**Comment by Erik262** [comment]
@j4hangir and this issue will last for 4+ years.

**Comment by j4hangir** [comment]
@Erik262 Damn… I’m also a paid member and right now TG is completely unusable. I have to open messages one by one.

I don’t dare clicking that "reset notifications" button either, since I’d have to mute god knows how many groups!

**Comment by Erik262** [comment]
> [@Erik262](https://github.com/Erik262) Damn… I’m also a paid member and right now TG is completely unusable. I have to open messages one by one.
> 
> I don’t dare clicking that "reset notifications" button either, since I’d have to mute god knows how many groups!

I was in the same situation, nothing helped. I can’t believe those programmers are unaware of it, or they simply don’t use Telegram at all.

**Comment by j4hangir** [comment]
> > [@Erik262](https://github.com/Erik262) Damn… I’m also a paid member and right now TG is completely unusable. I have to open messages one by one.
> > I don’t dare clicking that "reset notifications" button either, since I’d have to mute god knows how many groups!
> 
> I was in the same situation, nothing helped. I can’t believe those programmers are unaware of it, or they simply don’t use Telegram at all.

Is Telegram going bankrupt or what? I thought it's doing pretty well!

**Comment by j4hangir** [comment]
Okay, good news then: I’ve figured out where the problem lies, and the workaround is pretty easy.

[NotificationService.swift#L816](https://github.com/TelegramMessenger/Telegram-iOS/blob/7504a0f92694a3ccbb544e05f97304f8d0891ba9/Telegram/NotificationService/Sources/NotificationService.swift#L816) is the only reference to `You have a new message`:

```swift

        let isLockedMessage: String?
        if let data = try? Data(contentsOf: URL(fileURLWithPath: appLockStatePath(rootPath: rootPath))), let state = try? JSONDecoder().decode(LockState.self, from: data), isAppLocked(state: state) {
            if let notificationsPresentationData = try? Data(contentsOf: URL(fileURLWithPath: notificationsPresentationDataPath(rootPath: rootPath))), let notificationsPresentationDataValue = try? JSONDecoder().decode(NotificationsPresentationData.self, from: notificationsPresentationData) {
                isLockedMessage = notificationsPresentationDataValue.applicationLockedMessageString
            } else {
                isLockedMessage = "You have a new message"
            }
        } else {
            isLockedMessage = nil
        }
```

It seemed to be a problem with `appLockState` being considered locked. Upon further investigation, I found it might stem from Telegram being locked with a passcode, a feature I’d never enabled but decided to toggle on and off to reset the state, and it worked. Notifications are back to normal on my end!

I suspect that, in my case, using **iPhone mirroring** triggered a bug that corrupted the `appLockState`.

**Comment by Erik262** [comment]
> Okay, good news then: I’ve figured out where the problem lies, and the workaround is pretty easy.
> 
> [NotificationService.swift#L816](https://github.com/TelegramMessenger/Telegram-iOS/blob/7504a0f92694a3ccbb544e05f97304f8d0891ba9/Telegram/NotificationService/Sources/NotificationService.swift#L816) is the only reference to `You have a new message`:
> 
>         let isLockedMessage: String?
>         if let data = try? Data(contentsOf: URL(fileURLWithPath: appLockStatePath(rootPath: rootPath))), let state = try? JSONDecoder().decode(LockState.self, from: data), isAppLocked(state: state) {
>             if let notificationsPresentationData = try? Data(contentsOf: URL(fileURLWithPath: notificationsPresentationDataPath(rootPath: rootPath))), let notificationsPresentationDataValue = try? JSONDecoder().decode(NotificationsPresentationData.self, from: notificationsPresentationData) {
>                 isLockedMessage = notificationsPresentationDataValue.applicationLockedMessageString
>             } else {
>                 isLockedMessage = "You have a new message"
>             }
>         } else {
>             isLockedMessage = nil
>         }
> It seemed to be a problem with `appLockState` being considered locked. Upon further investigation, I found it might stem from Telegram being locked with a passcode, a feature I’d never enabled but decided to toggle on and off to reset the state, and it worked. Notifications are back to normal on my end!
> 
> I suspect that, in my case, using **iPhone mirroring** triggered a bug that corrupted the `appLockState`.

Where did you toggle this? In telegram or in system settings?

**Comment by j4hangir** [comment]
@Erik262 on iOS app, the option is named "Passcode & Face ID" under Privacy and Security.

**Comment by nasaboy** [comment]
@j4hangir I tested it using your method, but it didn't work.

**Comment by phantasm** [comment]
This happened to me on my old iPhone SE 2020. For a few months, I was receiving a whole bunch of these messages every day. My phone was getting hot because of them, and the battery was draining very quickly.

Then I bought a new iPhone and restored it from an iCloud backup. The problem was gone for a whole week until today. Mother of God, not again. Please fix it, I'm getting mad already. Didn't use iPhone mirroring btw.

I'm also using desktop app, and when "You have a new message" appears on a phone, desktop app don't have any new unmuted messages. And I have a decent number of subscribed channels in archive which yield something like 1k+ messages per hour in total, maybe that's the case (but it's still a bug I think)

Rebooting the device fixes the problem for several hours or even 1-2 days, but it always comes back.

**Comment by zenoleg** [comment]
Same problem right now

**Comment by MadestaQ** [comment]
It’s been FOUR YEARS already, and the problem still hasn't been fixed. The quality of the bug fixing is just appalling, guys. The phone constantly overheats due to the flood of notifications, and I gather there are no plans to fix it - a real "dream team" at work.

**Comment by Osbornnnnn** [comment]
Why this bug cant be fixed ??????

**Comment by Erik262** [comment]
> Why this bug cant be fixed ??????

It can, devs just dont give a f*
